sqlobject/src/main/java/org/jdbi/v3/sqlobject/customizers/BindIn.java

```diff
@@ -132,9 +152,20 @@ public SqlStatementCustomizer createForParameter(final Annotation annotation, fi
         @Override
         public Binder<BindIn, Object> build(final BindIn bindIn)
         {
-            final String key = bindIn.value();
-
             return (q, param, index, bind, arg) -> {
+                String name = bindIn.value();
+                if (name.isEmpty()) {
+                    if (param.isNamePresent()) {
+                        name = param.getName();
```

Comment:
This logic duplicates `Bind`'s parameter handling -- probably worth pulling it out into a util class?  Seems likely we'll want to use it again.